source/manual/load-test.html.md

```diff
@@ -0,0 +1,116 @@
+---
+owner_slack: "#govuk-developers"
+title: Run a load test with k6
+section: Testing
+layout: manual_layout
+parent: "/manual.html"
+---
+
+These instructions are based on [this blog post](https://grafana.com/blog/2022/06/23/running-distributed-load-tests-on-kubernetes/).
+
+1. Log into AWS and set the context to the environment you wish to test:
+
+    ```sh
+    ENVIRONMENT=staging
+    ROLE=admin
+    AWS_REGION=eu-west-1
+
+    eval $(gds aws govuk-${ENVIRONMENT}-${ROLE} -e --art 8h)
+    kubectl config use-context $ENVIRONMENT
+    ```
+
+    > You will need to assume the `admin` role in order to deploy k6 to the cluster.
+
+1. Clone the `k6-operator` repo and make this:
+
+    ```sh
+    git clone https://github.com/grafana/k6-operator && cd k6-operator
+    make deploy
+    ```
+
+    > If you get an error similar to `invalid go version '1.22.0': must match format 1.23` then you need to upgrade `go` to the correct version.
+    > This can be done by using Homebrew, e.g. `brew install go@1.23`
+
+1. Write your test script, for example to test 3000 virtual users against the Collections app over 30 seconds (bypassing router):
+
+    ```javascript
+    import http from 'k6/http';
+    import { check } from 'k6';
+
+    export const options = {
+      vus: 3000,
+      duration: '30s',
+    }
+
+    export default function() {
+      const res = http.get('http://collections.apps.svc.cluster.local/world');
+
+      check(res, {
+        'is status 200': (r) => r.status === 200,
+        'is status 2xx, not 200': (r) => r.status >= 201 && r.status <= 299,
+        'is status 3xx': (r) => r.status >= 300 && r.status <= 399,
+        'is status 4xx': (r) => r.status >= 400 && r.status <= 499,
+        'is status 5xx': (r) => r.status >= 500 && r.status <= 599,
+      });
+    }
+
+    ```
+
+1. Run your code locally to make sure it works:
+
+    ```sh
+    k6 run test.js
+    ```
+
+    > You may see errors related to connections as you may not be able to establish the correct number of connections from your machine.
+
+1. Deploy the test script:
+
+    ```sh
+    kubectl create configmap my-test-name --from-file /path/to/our/test.js
```

Comment:
**suggestion:** naming guidance

What's the meaning of the test name here? Is it something that will appear in some interface? Is there anything to bear in mind when choosing a name?

Reply:
It's just a name that is used throughout to refer to the configmap (and therefore the name of the pods).  I'll add a note about this.